Model/Component/Rewrites.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace CtiDigital\Configurator\Model\Component;
+
+use Magento\Framework\ObjectManagerInterface;
+use Magento\UrlRewrite\Model\UrlRewriteFactory;
+use Magento\UrlRewrite\Model\UrlPersistInterface;
+use CtiDigital\Configurator\Model\LoggingInterface;
+use CtiDigital\Configurator\Model\Exception\ComponentException;
+
+class Rewrites extends CsvComponentAbstract
+{
+    protected $alias = 'rewrites';
+    protected $name = 'rewrites';
+    protected $description = 'Component to create URL Store Rewrites';
+
+    /**
+     * @var UrlPersistInterface
+     */
+    protected $urlPersist;
+
+    /**
+     * @var UrlRewriteFactory
+     */
+    protected $urlRewriteFactory;
+
+    /**
+     * Rewrites constructor.
+     * @param LoggingInterface $log
+     * @param ObjectManagerInterface $objectManager
+     * @param UrlPersistInterface $urlPersist
+     * @param UrlRewriteFactory $urlRewriteFactory
+     */
+    public function __construct(
+        LoggingInterface $log,
+        ObjectManagerInterface $objectManager,
+        UrlPersistInterface $urlPersist,
+        UrlRewriteFactory $urlRewriteFactory
+    ) {
+        parent::__construct($log, $objectManager);
+        $this->urlPersist = $urlPersist;
+        $this->urlRewriteFactory = $urlRewriteFactory;
+
+    }
+
+    /**
+     * @param array|null $data
+     */
+    protected function processData(array $data = null)
+    {
+        // Get the first row of the CSV file for the attribute columns.
+        if (!isset($data[0])) {
+            throw new ComponentException(
+                sprintf('The row data is not valid.')
+            );
+        }
+
+        $attributeKeys = $this->getAttributesFromCsv($data);
+        unset($data[0]); //Remove columns instead of skipping within foreach
+
+        foreach ($data as $rewriteData) {
```

Comment:
This loop contains more levels of nesting that I like. Is it possible to extract the contents into a separate method?